src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -112,6 +127,25 @@ private void EndRequest(HttpContext httpContext, long timestamp)
                     telemetry.Name = httpContext.Request.Method + " " + httpContext.Request.Path.Value;
                 }
 
+                if (string.IsNullOrEmpty(telemetry.Context.InstrumentationKey))
+                {
+                    this.client.Initialize(telemetry);
```

Comment:
I've changed this so that the telemetry is initialized just before the call to TrackRequest(). I'm not sure why the other implementations of HostingDiagnosticListener (dotnet-server) do this, but I'm just trying to follow their pattern.